src/AppBundle/Subscriber/StatusChangeOnPushSubscriber.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace AppBundle\Subscriber;
+
+use AppBundle\Event\GitHubEvent;
+use AppBundle\GitHubEvents;
+use AppBundle\Issues\Status;
+use AppBundle\Issues\StatusApi;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Changes the status from "Needs Work" to "Needs Review" when pushing new 
+ * commits.
+ *
+ * @author Wouter de Jong <wouter@wouterj.nl>
+ */
+class StatusChangeOnPushSubscriber implements EventSubscriberInterface
+{
+    private $statusApi;
+
+    public function __construct(StatusApi $statusApi)
+    {
+        $this->statusApi = $statusApi;
+    }
+
+    public function onPullRequest(GitHubEvent $event)
+    {
+        $data = $event->getData();
+        if ('synchronize' !== $data['action']) {
+            $event->setResponseData(array('unsupported_action' => $data['action']));
+
+            return;
+        }
+
+        $repository = $event->getRepository();
+        $issueNumber = $data['issue']['number'];
+        $responseData = array('issue' => $issueNumber);
+        $status = $this->statusApi->getIssueStatus($issueNumber, $repository);
+
+        if (Status::NEEDS_REVIEW === $status) {
+            $responseData['status_change'] = null;
+            $event->setResponseData($responseData);
+
+            return;
+        }
```

Comment:
Perhaps we should only change the status if it's currently in the `NEEDS_WORK` status? For example, if something is already `REVIEWED`, it's possible this was a tiny fix that a maintainer asked for, and it should really stay in `REVIEWED`.